src/main/scala/nodes/learning/ApproximatePCA.scala

```diff
@@ -62,10 +64,11 @@ object ApproximatePCAEstimator {
    * @param q Number of iterations to use in the computation.
    * @return An n x (k+p) matrix that approximates A
    */
-  def approximateQ(A: DenseMatrix[Double], l: Int, q: Int): DenseMatrix[Double] = {
+  def approximateQ(A: DenseMatrix[Double], l: Int, q: Int, seed: Int = 0): DenseMatrix[Double] = {
     val d = A.cols
 
-    val omega = new DenseMatrix(d, l, randn(d*l).toArray) //cpu: d*l, mem: d*l
+    implicit val randBasis: RandBasis = new RandBasis(new ThreadLocalRandomGenerator(new MersenneTwister(seed)))
```

Comment:
any reason why this is marked implicit even though you manually pass it in the next line?